comp/core/hostname/hostnameimpl/service.go

```diff
@@ -43,7 +43,7 @@ func (hs *service) GetWithProvider(ctx context.Context) (pkghostname.Data, error
 	return pkghostname.GetWithProvider(ctx)
 }
 
-// newHostnameService fetches the hostname and returns a service wrapping it
-func newHostnameService() hostname.Component {
+// NewHostnameService fetches the hostname and returns a service wrapping it
+func NewHostnameService() hostname.Component {
```

Comment:
This can be reverted back to the non-public constructor. We depend on FX to resolve and provide instances of the component based on the [module](https://github.com/DataDog/datadog-agent/blob/723613ce7248a0d2b01d902d21db16073257d9b8/comp/core/hostname/hostnameimpl/service.go#L18).

Reply:
I had to make this public again in order to pass in the hostname implementation to usages of `pipeline.NewProvider` in `pkg/` (non fx components).

Reply:
We should not make this contractor public on it's own, as it being private (non exported) ensure proper use of components across the agent. Another reason to not make this public is, it makes very difficult to add dependencies to this component in the future since we can non longer rely on FX to fulfill them since its called elsewhere in the agent codebase. 

We have a pattern for working around this for non-fx apps (that have not yet, but will soon be converted). The pattern is to supply a second public constructor that explicitly calls out it's purpose, [like this](https://github.com/DataDog/datadog-agent/blob/ff99122acb83f841a058d4ed7ba59f76e73a862e/comp/dogstatsd/server/server.go#L193).